packages/ember-data/lib/system/model/model.js

```diff
@@ -473,12 +473,31 @@ var Model = Ember.Object.extend(Ember.Evented, {
   },
 
   _setup: function() {
+    var record = this;
     this._changesToSync = {};
     this._deferredTriggers = [];
     this._data = {};
     this._attributes = Ember.create(null);
     this._inFlightAttributes = Ember.create(null);
-    this._relationships = {};
+    this._relationships = Ember.create({
```

Comment:
I think we should move this out to its own `Relationships` or similar function / prototype. Otherwise this will begin to create more memory than it needs to as each object will have its own copy of the function.